Birthday-2024-Project/Scripts/GameManager.gd

```diff
@@ -33,13 +33,51 @@ var _can_interact: bool
 var _current_state: GameState
 var _is_inititialized: bool
 var _previous_state: GameState
+var _gm : GameMaster
+var _levelData : LevelSetup
 
 signal initialized_event()
 signal state_changed_event(state)
 
 func go_to_main_menu():
-	myScreen.GoToScreen(load("res://MainScenes/main_menu.tscn"), {}, ScreenManager.TransitionStyle.BACK_PAGE)
-	pass
+	var thisLevelIndex : int = myScreen.transitionData[LevelsSelectMenu.PASS_LEVEL_INDEX_KEY]
+	var buttonsPerPage : int = myScreen.transitionData[LevelsSelectMenu.BUTTONS_PER_PAGE_KEY]
+	var pageNum : int = 1 + floori(thisLevelIndex / buttonsPerPage)
+	
+	if myScreen.transitionData[LevelsSelectMenu.IS_CAMPAIGN_KEY]:
+		myScreen.GoToScreen(load(str(LevelsSelectMenu.CAMPAIGN_LEVELS, pageNum, ".tscn")), {}, ScreenManager.TransitionStyle.BACK_PAGE)
+	else:
+		myScreen.GoToScreen(load(str(LevelsSelectMenu.SAPLING_LEVELS, pageNum, ".tscn")), {}, ScreenManager.TransitionStyle.BACK_PAGE)
+
+func go_to_next_level():
+	var thisLevelIndex : int = myScreen.transitionData[LevelsSelectMenu.PASS_LEVEL_INDEX_KEY]
+	var transitionData : Dictionary = {}
+	transitionData[LevelsSelectMenu.PASS_LEVEL_INDEX_KEY] = thisLevelIndex + 1
+	var buttonsPerPage : int = myScreen.transitionData[LevelsSelectMenu.BUTTONS_PER_PAGE_KEY]
+	transitionData[LevelsSelectMenu.BUTTONS_PER_PAGE_KEY] = buttonsPerPage
+	
+	if myScreen.transitionData[LevelsSelectMenu.IS_CAMPAIGN_KEY]:
+		#last level
+		if thisLevelIndex + 1 >= _gm.campaign_level_library.Levels.size():
```

Comment:
Unless I'm mistaken, it's actually because there campaign levels library currently consists of 5 instances of TestLevel. So even if you progress or skip, you just see the same level but it's technically the next level